3. PRODUCT ANALYSIS

1. BUSINESS UNDERSTANDING

Perusahaan ingin memahami karakteristik produk untuk mengetahui:

- Produk yang sering dibeli oleh pelanggan.
- Kategori produk dengan nilai transaksi tertinggi.
- Kontribusi dari masing masing sub kategori produk.

2. DATA UNDERSTANDING

In [123]:
import pandas as pd

In [124]:
df = pd.read_csv(r"D:\PROJECT\Superstore Sales Dataset\Dataset\Superstore Sales Dataset.csv")
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales
0,1,CA-2017-152156,8/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600
1,2,CA-2017-152156,8/11/2017,11/11/2017,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.0,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400
2,3,CA-2017-138688,12/6/2017,16/06/2017,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036.0,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200
3,4,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775
4,5,US-2016-108966,11/10/2016,18/10/2016,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.0,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680


In [125]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9800 entries, 0 to 9799
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9800 non-null   int64  
 1   Order ID       9800 non-null   object 
 2   Order Date     9800 non-null   object 
 3   Ship Date      9800 non-null   object 
 4   Ship Mode      9800 non-null   object 
 5   Customer ID    9800 non-null   object 
 6   Customer Name  9800 non-null   object 
 7   Segment        9800 non-null   object 
 8   Country        9800 non-null   object 
 9   City           9800 non-null   object 
 10  State          9800 non-null   object 
 11  Postal Code    9789 non-null   float64
 12  Region         9800 non-null   object 
 13  Product ID     9800 non-null   object 
 14  Category       9800 non-null   object 
 15  Sub-Category   9800 non-null   object 
 16  Product Name   9800 non-null   object 
 17  Sales          9800 non-null   float64
dtypes: float

In [126]:
df.shape

(9800, 18)

In [127]:
print(df.columns)

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales'],
      dtype='object')


-> Data set memiliki 9800 data dengan 18 kolom, dengan tipe data yg berbeda terutama untuk kolom Postal Code dan Sales memiliki tipe data float64, yang lain berupa object.

3. DATA PREPARATION

In [128]:
df.isnull().sum()

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country           0
City              0
State             0
Postal Code      11
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
dtype: int64

In [129]:
df.duplicated().sum()

0

-> Sesuai dengan pengamatan sebelumnya bahwa data yang memiliki nilai nol hanya di kolom Postal Code dan data kembar tidak ada.

4. Product Overview

In [130]:
total_kategori_produk = df["Category"].nunique()
total_sub_produk = df["Sub-Category"].nunique()
total_produk = df["Product Name"].nunique()

print(f"Total Kategori Produk       : {total_kategori_produk}")
print(f"Total Sub Kategori Produk   : {total_sub_produk}")
print(f"Total Produk                : {total_produk}")

Total Kategori Produk       : 3
Total Sub Kategori Produk   : 17
Total Produk                : 1849


-> Pada dataset kali ini perusahaan telah menjual sebanyak 1849 macam produk yang terbagi dalam 17 sub produk dan terbagi menjadi 3 kategori produk

5. Category Product Analysis

In [131]:
kategoriproduk_sales = (
    df.groupby(["Category"]) ["Sales"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

kategoriproduk_order = (
    df.groupby(["Category"]) ["Order ID"]
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)

kategoriproduk_all = pd.merge(
    kategoriproduk_sales,
    kategoriproduk_order,
    on=["Category"]
)

kategoriproduk_all.head()

,Category,Sales,Order ID
0,Technology,827455.8730,1519
1,Furniture,728658.5757,1727
2,Office Supplies,705422.3340,3676


In [132]:
kategoriproduk_display = kategoriproduk_all.copy()

kategoriproduk_display["Sales"] = kategoriproduk_display["Sales"].map('${:,.2f}'.format)

kategoriproduk_display

,Category,Sales,Order ID
0,Technology,"$827,455.87",1519
1,Furniture,"$728,658.58",1727
2,Office Supplies,"$705,422.33",3676


-> Produk yang terjual terdapat 3 kategori yaitu Technology, Furniture, dan Office Supplies

-> Dari 3 kategori sales terbesar ada pada kategori Technology tetapi memiliki penjualan yang sangat sedikit

-> sedangkan untuk kategori office supplies memiliki penjualan terkecil tetapi memiliki penjualan yang terbanyak bahkan 2 kali lipat lebihh dari kategori technology

Bisa disimpulkan bahwa kategori Technology merupakan kategori dengan total penjualan terbesar dengan nilai $827.455,87, walaupun hanya mencatat sebanyak 1.519 transaksi. 
Sedangkan, Office Supplies memiliki jumlah transaksi tertinggi yaitu 3.676 order, tetapi total penjualannya merupakan yang paling rendah dibanding kategori lainnya. Hal ini menunjukkan bahwa pelanggan cenderung membeli produk Office Supplies lebih sering, sementara transaksi pada kategori Technology memiliki nilai pembelian yang lebih besar pada setiap order.

6. Sub Category Product Analysis

In [133]:
subkategoriproduk_sales = (
    df.groupby(["Sub-Category"]) ["Sales"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

subkategoriproduk_order = (
    df.groupby(["Sub-Category"]) ["Order ID"]
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)

subkategoriproduk_all = pd.merge(
    subkategoriproduk_sales,
    subkategoriproduk_order,
    on=("Sub-Category")
)

subkategoriproduk_all


,Sub-Category,Sales,Order ID
0,Phones,327782.4480,803
1,Chairs,322822.7310,566
2,Storage,219343.3920,764
3,Tables,202810.6280,302
4,Binders,200028.7850,1291
5,Machines,189238.6310,112
6,Accessories,164186.7000,702
7,Copiers,146248.0940,66
8,Bookcases,113813.1987,222
9,Appliances,104618.4030,444


In [134]:
subkategoriproduk_display = subkategoriproduk_all.copy()

subkategoriproduk_display["Sales"] = subkategoriproduk_display["Sales"].map('${:,.2f}'.format)

subkategoriproduk_display

,Sub-Category,Sales,Order ID
0,Phones,"$327,782.45",803
1,Chairs,"$322,822.73",566
2,Storage,"$219,343.39",764
3,Tables,"$202,810.63",302
4,Binders,"$200,028.79",1291
5,Machines,"$189,238.63",112
6,Accessories,"$164,186.70",702
7,Copiers,"$146,248.09",66
8,Bookcases,"$113,813.20",222
9,Appliances,"$104,618.40",444


-> Sub Kategori produk yang terjual dengan nilai sales yang terbanyak ada di Phone dengan nilai sebesar $327,782.45 dengan total penjualan sebanyak 803

-> Sub Kategori produk yang terjual nya paling banyak ada di Binders total 1.291 penjualan dengan nilai sales sebesar $202,810.63

-> Sub Kategori produk yang terjual dengan nilai sales yang terkecil ada di Fasteners dengan nilai sebesar $3,001.96 dengan total penjualan sebanyak 212

-> Sub Kategori produk yang terjual nya paling rendah ada di Copiers total 66 penjualan dengan nilai sales sebesar $146,248.09


7. PRODUCT ANALYSIS

In [135]:
produk_sales = (
    df.groupby(["Product Name", "Sub-Category", "Category"]) ["Sales"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

produk_order = (
    df.groupby(["Product Name","Sub-Category", "Category"]) ["Order ID"]
    .nunique()
    .sort_values(ascending=False)
    .reset_index()
)

produksales_all = pd.merge(
    produk_sales,
    produk_order,
    on=["Product Name","Sub-Category", "Category"]
)

produkorder_all = pd.merge(
    produk_order,
    produk_sales,
    on=["Product Name", "Sub-Category", "Category"]
)

In [136]:
produkorder_all.head(10)

,Product Name,Sub-Category,Category,Order ID,Sales
0,Staple envelope,Envelopes,Office Supplies,47,1675.652
1,Staples,Fasteners,Office Supplies,46,755.470
2,Easy-staple paper,Paper,Office Supplies,44,2414.156
3,Avery Non-Stick Binders,Binders,Office Supplies,20,217.316
4,Staple remover,Supplies,Office Supplies,18,263.088
5,Staples in misc. colors,Art,Office Supplies,18,459.660
6,KI Adjustable-Height Table,Tables,Furniture,17,4466.661
7,Storex Dura Pro Binders,Binders,Office Supplies,17,278.586
8,Staple-based wall hangings,Furnishings,Furniture,16,422.288
9,"Situations Contoured Folding Chairs, 4/Set",Chairs,Furniture,15,3641.274


In [140]:
produk_display = produkorder_all.copy()

produk_display["Sales"] = produk_display["Sales"].map('${:,.2f}'.format)

produk_display

,Product Name,Sub-Category,Category,Order ID,Sales
0,Staple envelope,Envelopes,Office Supplies,47,"$1,675.65"
1,Staples,Fasteners,Office Supplies,46,$755.47
2,Easy-staple paper,Paper,Office Supplies,44,"$2,414.16"
3,Avery Non-Stick Binders,Binders,Office Supplies,20,$217.32
4,Staple remover,Supplies,Office Supplies,18,$263.09
...,...,...,...,...,...
1844,Avery Hi-Liter Pen Style Six-Color Fluorescent...,Art,Office Supplies,1,$7.70
1845,Penpower WorldCard Pro Card Scanner,Machines,Technology,1,$91.47
1846,4009 Highlighters,Art,Office Supplies,1,$8.04
1847,LG G2,Phones,Technology,1,"$1,499.97"


-> Produk Staple Evelope adalah produk yang sering di beli oleh customer sebanyak 47 dengan nilai penjualan sebesar $1,675.65


In [138]:
produksales_all.head(10)

,Product Name,Sub-Category,Category,Sales,Order ID
0,Canon imageCLASS 2200 Advanced Copier,Copiers,Technology,61599.824,5
1,Fellowes PB500 Electric Punch Plastic Comb Bin...,Binders,Office Supplies,27453.384,10
2,Cisco TelePresence System EX90 Videoconferenci...,Machines,Technology,22638.480,1
3,HON 5400 Series Task Chairs for Big and Tall,Chairs,Furniture,21870.576,8
4,GBC DocuBind TL300 Electric Binding System,Binders,Office Supplies,19823.479,11
5,GBC Ibimaster 500 Manual ProClick Binding System,Binders,Office Supplies,19024.500,9
6,Hewlett Packard LaserJet 3310 Copier,Copiers,Technology,18839.686,8
7,HP Designjet T520 Inkjet Large Format Printer ...,Machines,Technology,18374.895,3
8,GBC DocuBind P400 Electric Binding System,Binders,Office Supplies,17965.068,6
9,High Speed Automatic Electric Letter Opener,Supplies,Office Supplies,17030.312,3


In [139]:
produksales_display = produksales_all.copy()

produksales_display["Sales"] = produksales_display["Sales"].map('${:,.2f}'.format)

produksales_display.head(10)

,Product Name,Sub-Category,Category,Sales,Order ID
0,Canon imageCLASS 2200 Advanced Copier,Copiers,Technology,"$61,599.82",5
1,Fellowes PB500 Electric Punch Plastic Comb Bin...,Binders,Office Supplies,"$27,453.38",10
2,Cisco TelePresence System EX90 Videoconferenci...,Machines,Technology,"$22,638.48",1
3,HON 5400 Series Task Chairs for Big and Tall,Chairs,Furniture,"$21,870.58",8
4,GBC DocuBind TL300 Electric Binding System,Binders,Office Supplies,"$19,823.48",11
5,GBC Ibimaster 500 Manual ProClick Binding System,Binders,Office Supplies,"$19,024.50",9
6,Hewlett Packard LaserJet 3310 Copier,Copiers,Technology,"$18,839.69",8
7,HP Designjet T520 Inkjet Large Format Printer ...,Machines,Technology,"$18,374.90",3
8,GBC DocuBind P400 Electric Binding System,Binders,Office Supplies,"$17,965.07",6
9,High Speed Automatic Electric Letter Opener,Supplies,Office Supplies,"$17,030.31",3


-> Produk Canon imageCLASS 2200 Advanced Copier adalah produk yang memiliki nilai penjualan terbesar yaitu $61,599.82, dengan total order sebanyak 5 kali